r-package/grf/R/lm_forest.R

```diff
@@ -0,0 +1,354 @@
+#'  For a detailed description of honesty, honesty.fraction, honesty.prune.leaves, and recommendations for
+#'  parameter tuning, see the grf algorithm reference.
+#' @param honesty.fraction The fraction of data that will be used for determining splits if honesty = TRUE. Corresponds
+#'                         to set J1 in the notation of the paper. Default is 0.5 (i.e. half of the data is used for
+#'                         determining splits).
+#' @param honesty.prune.leaves If TRUE, prunes the estimation sample tree such that no leaves
+#'  are empty. If FALSE, keep the same tree as determined in the splits sample (if an empty leave is encountered, that
+#'  tree is skipped and does not contribute to the estimate). Setting this to FALSE may improve performance on
+#'  small/marginally powered data, but requires more trees (note: tuning does not adjust the number of trees).
+#'  Only applies if honesty is enabled. Default is TRUE.
+#' @param alpha A tuning parameter that controls the maximum imbalance of a split. Default is 0.05.
+#' @param imbalance.penalty A tuning parameter that controls how harshly imbalanced splits are penalized. Default is 0.
+#' @param stabilize.splits Whether or not Wk should be taken into account when
+#' determining the imbalance of a split. It is an exact extension of the single-arm constraints (detailed
+#' in the causal forest algorithm reference) to multiple arms, where the constraints apply to each regressor Wk.
+#' Default is FALSE.
+#' @param ci.group.size The forest will grow ci.group.size trees on each subsample.
+#'                      In order to provide confidence intervals, ci.group.size must
+#'                      be at least 2. Default is 2. (Confidence intervals are
+#'                      currently only supported for univariate outcomes Y).
+#' @param compute.oob.predictions Whether OOB predictions on training set should be precomputed. Default is TRUE.
+#' @param num.threads Number of threads used in training. By default, the number of threads is set
+#'                    to the maximum hardware concurrency.
+#' @param seed The seed of the C++ random number generator.
+#'
+#' @return A trained lm forest object.
+#'
+#' @examples
+#' \donttest{
+#' # Train a LM Forest to estimate CATEs in a regression discontinuity design.
+#' # Simulate a simple example with a heterogeneous jump in the CEF.
+#' n <- 2000
+#' p <- 5
+#' X <- matrix(rnorm(n * p), n, p)
+#' Z <- runif(n, -4, 4)
+#' cutoff <- 0
+#' W <- as.numeric(Z >= cutoff)
+#' tau <- pmax(0.5 * X[, 1], 0)
+#' Y <- tau * W  + 1 / (1 + exp(2 * Z)) + 0.2 * rnorm(n)
+#'
+#' # Compute the Imbens-Kalyanaraman MSE-optimal bandwidth for a local linear regression.
+#' if (require("rdd", quietly = TRUE)) {
+#'   bandwidth <- IKbandwidth(Z, Y, cutoff)
+#'   # Compute kernel weights for a triangular kernel.
+#'   sample.weights <- kernelwts(Z, cutoff, bandwidth, "triangular")
+#' } else {
```

Comment:
I don't think we should use an arbitrary BW here, even though it's an example. How about making the loading of rdd non-optional for running this example, then add a comment showing how to alternatively hand-set the BW:

bandwidth <- # user can hand-specify this
dist <- abs((Z - cutoff) / bandwidth)
sample.weights <- (1 - dist) * (dist <= 1) / bandwidth